.travis.yml

```diff
@@ -0,0 +1,46 @@
+before_install:
+  - if test -e $HOME/miniconda/bin; then
+        echo "miniconda already installed.";
+    else
+      if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
+        wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O miniconda.sh;
+      else
+        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+      fi
+
+      rm -rf $HOME/miniconda;
+      bash miniconda.sh -b -p $HOME/miniconda;
+      export PATH="$HOME/miniconda/bin:$PATH";
+      hash -r;
+      conda config --set always_yes yes --set changeps1 no;
+      conda update -q conda;
+
+      conda info -a;
+
+      conda config --add channels defaults;
+      conda config --add channels conda-forge;
+      conda config --add channels bioconda;
```

Comment:
So, you need to use an environment creation step here- as per https://github.com/ebi-gene-expression-group/r-seurat-scripts/blob/master/.travis.yml. Have the Scanpy install happen there instead of where you have it now. Then it will be part of the caching setup.